crm_timesheet/models/crm_phonecall.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# License AGPL-3 - See LICENSE file on root folder for details
+##############################################################################
+
+from openerp import models, fields, api
+from openerp.fields import DATE_LENGTH
+from datetime import datetime
+from openerp.tools.translate import _
+
+
+class CrmPhonecall(models.Model):
+    _inherit = 'crm.phonecall'
+
+    analytic_account = fields.Many2one(
+        comodel_name='account.analytic.account')
+    timesheets = fields.One2many(comodel_name='hr.analytic.timesheet',
+                                 inverse_name='phonecall')
+
+    def _timesheet_prepare(self, vals):
+        """
+            NOTE: Call this method ONLY from api.model or api.one methods
+
+            param: vals
+            type: dict
+            desc: crm.phonecall vals dict (It's like create and write methods)
+        """
+        def value_or_id(val):
```

Comment:
This method is not needed because you know each field if is a relational of regular field.